### DISCUSSION TITLE
Need an error when choosing a wrong controller?

### DISCUSSION THREAD

**Discussion by amrytov** [cyberbotics/webots#5464] [Ideas]
Hi,

This is not an important thing but still may be worth noting (and eventually fixing):

In an educations scene I use a PROTO robot, in which controller names are restricted like this:

      field  SFString{"code_py", "code_c"}  controller "code_c"

The controllers are some unfinished code templates for the students to complete, in their language of choice.  
This works perfectly.  

However,  there is still the [Select...] button below, which knows nothing about the restrictions
and shows a whole bunch of other controllers the students should not even see.  The good thing is when a student
chooses a controller which is not in the allowed list, nothing happens (the controller field does not change).  
The bad thing is that there is absolutely no indication that something wrong has happened.  In similar situations when one tries to enter a wrong field value, there is usually an error message in the console. 

Would be good to add one of the following:

 - Perfect:  if there is a restriction list on the 'controller' field, the "Choose..." button is hidden or disabled
 - Less perfect:  in such situation the "Choose..." button works, but shows only the allowed values
 - Somewhat ok:  no change in the "Choose...",  but at least report an error ("wrong value") to console when trying to assign an invalid value.

**Comment by omichel** [comment]
This is a very good point. I will move it to an issue so that we can implement it at some point. Again, this should be fairly easy to implement and you are welcome to do so. We would be delighted to review your PR about it.

**Reply by omichel** [reply]
See #5466.

**Reply by amrytov** [reply]
I'm desperately trying to keep up with the deadlines at the moment, once this is over I might come up with a PR or two...  Diving deep into WeBots code feels very tempting...
